source/fab/dep_tree.py

```diff
@@ -23,20 +24,30 @@ class AnalysedFile(object):
 
     """
 
-    def __init__(self, fpath: Path, file_hash, symbol_deps=None, symbol_defs=None, file_deps=None,
+    def __init__(self, fpath, file_hash, module_defs=None, symbol_defs=None, symbol_deps=None, file_deps=None,
```

Comment:
Indeed. We've been shying away from that because it bloats the code so much, and reduces readability (and is so ugly), but it _is_ more useful. Having thought about it again now, we'll switch back to using it and will change it here. Roll on 3.10 as a minimum version so we can use | instead.